src/log/log.c

```diff
@@ -2227,30 +2226,21 @@ __log_write_internal(WT_SESSION_IMPL *session, WT_ITEM *record, WT_LSN *lsnp,
 		if (free_slot)
 			__wt_log_slot_free(session, myslot.slot);
 	} else if (force) {
-		/*
-		 * If we are going to wait for this slot to get written,
-		 * signal the wrlsn thread.
-		 *
-		 * XXX I've seen times when conditions are NULL.
-		 */
-		if (conn->log_cond != NULL) {
-			__wt_cond_signal(session, conn->log_cond);
-			__wt_yield();
-		} else
-			WT_ERR(__wt_log_force_write(session, 1, NULL));
+		__wt_cond_signal(session, &conn->log_cond);
+		__wt_yield();
```

Comment:
`wt_logmgr_create` is always called before recovery is run.  So logging is initialized.  It goes:
* `wt_logmgr_create` to initialize the subsystem (including
* Run recovery
* `wt_logmgr_open` to start worker threads

I misspoke - reconfigure does not do anything with the cond variables and does not come through this path.  They already get created/destroyed as you suggest, created during open and destroyed on close.  I wonder if this is leftover from the earlier logging code because when I add an fatal assertion in the original else clause it never triggers, even running test/recovery and the recovery tests in the test suite.